pep-0484.txt

```diff
@@ -276,6 +276,34 @@ implemented by deferring to ``isinstance(x, collections.abc.Callable)``.
 However, ``isinstance(x, typing.Callable[...])`` is not supported.
 
 
+Decorators
+----------
+
+Decorators can modify the types of the functions or classes they
+decorate. Use the ``decorated_type`` decorator to declare the type of
+the resulting item after all other decorators have been applied::
+
+  from typing import ContextManager, Iterator, decorated_type
+  from contextlib import contextmanager
+
+  class DatabaseSession: ...
+
+  @decorated_type(Callable[[str], ContextManager[DatabaseSession]])
+  @contextmanager
```

Comment:
May need a different example than `contextmanager` since (at least mypy) now gets the type right there through the use of plugins.